crates/swc_ecma_utils/src/lib.rs

```diff
@@ -411,6 +411,12 @@ pub struct Hoister {
 impl Visit for Hoister {
     noop_visit_type!();
 
+    fn visit_assign_pat_prop(&mut self, node: &AssignPatProp, _: &dyn Node) {
+        node.value.visit_with(node, self);
+
+        self.vars.push(node.key.clone());
```

Comment:
Does this `Hoister` need to do the `is_pat_decl` stuff that `BindingCollector` is doing?

Reply:
I don' think so because `visit_pat` does not check if it's decl.